aiida/cmdline/commands/cmd_verdi.py

```diff
@@ -33,6 +34,23 @@
 )
 
 
+def _start_of_option(value: str) -> bool:
+    """Check if the value looks like the start of an option.
+
+    This is an adaptation of :py:func:`click.shell_completion._start_of_option` that simply add ``.`` and ``~`` as the
+    characters that are interpreted as the start of a filepath, and so not the start of an option. This will ensure that
+    filepaths starting with these characters are autocompleted once again.
+    """
+    if not value:
+        return False
+
+    # Allow characters that typically designate the start of a path.
+    return not value[0].isalnum() and value[0] not in ['/', '.', '~']
```

Comment:
```suggestion
    return not value[0].isalnum() and value[0] not in ['/', '.', '~', '$']
```

I think should also add to escape for `$HOME` and `$PWD` here.